src/components/inspectors/SignalSelect.vue

```diff
@@ -0,0 +1,360 @@
+    },
+    validateName(name) {
+      if (!name) {
+        return this.$t('Signal Name is required');
+      }
+      return '';
+    },
+    confirmDeleteSignal() {
+      this.showConfirmDelete = false;
+      const index = window.ProcessMaker.$modeler.definitions.rootElements.findIndex(element => element.id === this.deleteSignal.id);
+      if (index) {
+        window.ProcessMaker.$modeler.definitions.rootElements.splice(index, 1);
+      }
+    },
+    removeSignal(signal) {
+      this.showConfirmDelete = true;
+      this.deleteSignal = signal;
+    },
+    toggleConfigSignal() {
+      this.showListSignals = !this.showListSignals;
+    },
+    editSignal(signal) {
+      this.signalId = signal.id;
+      this.signalName = signal.name;
+      this.showEditSignal = true;
+    },
+    getSignalById(id) {
+      return window.ProcessMaker.$modeler.definitions.rootElements.find(element => element.id === id);
+    },
+    change(value) {
+      if (!value) {
+        this.$emit('input', '');
+        this.fixUnrefreshedVueFormRender();
+        return;
+      }
+      let signal = this.getSignalById(value.id);
+      if (!signal) {
+        signal = window.ProcessMaker.$modeler.moddle.create('bpmn:Signal', {
+          id: value.id,
+          name: value.name,
+        });
+        window.ProcessMaker.$modeler.definitions.rootElements.push(signal);
+      }
+      this.$emit('input', this.storeId ? get(value, this.trackBy) : value);
+      this.fixUnrefreshedVueFormRender();
+    },
+    fixUnrefreshedVueFormRender() {
+      this.$nextTick(() => {
+        let form = this;
+        while (form && form.$options._componentTag !== 'vue-form-renderer') {
+          form = form.$parent;
+        }
+        if (form) {
+          form.$emit('update', form.transientData);
+        }
+      });
+    },
+    showAddSignal() {
+      this.showNewSignal = true;
+      this.signalId = '';
+      this.signalName = '';
+    },
+    cancelAddSignal() {
+      this.showNewSignal = false;
+      this.showEditSignal = false;
+    },
+    updateSignal() {
+      this.getSignalById(this.signalId).name = this.signalName;
+      this.showEditSignal = false;
+    },
+    addSignal() {
+      const signal = window.ProcessMaker.$modeler.moddle.create('bpmn:Signal', {
+        id: this.signalId,
+        name: this.signalName,
+      });
+      window.ProcessMaker.$modeler.definitions.rootElements.push(signal);
+      this.showNewSignal = false;
+    },
+    updateOptions() {
+      this.options = this.globalSignals;
+      window.ProcessMaker.$modeler.definitions.rootElements.forEach((element) => {
+        const localSignal = this.options.find(option => option.id === element.id);
+        if (element.$type === 'bpmn:Signal' &&!localSignal) {
+          this.options.push({
+            id: element.id,
+            name: element.name,
+          });
+        } else if (localSignal) {
+          localSignal.name = element.name;
+        }
+      });
+    },
```

Comment:
If you take advantage of [`uniqBy`](https://lodash.com/docs/4.17.15#uniqBy), then you can simplify this block:

```suggestion
    updateOptions(globalOptions) {
      this.options = uniqBy([...this.localSignals, ...globalOptions], 'id');
    },
```

**Note:** The order of the arrays passed to `uniqBy` is important. The fist duplicate item it encounters will be the one it returns in the results; by specifying `localSignals` first, it captures the updated signal name.

Reply:
Beautiful, nice code, changed.